SimplenoteWidgets/Providers/NoteWidgetProvider.swift

```diff
@@ -0,0 +1,92 @@
+import WidgetKit
+
+struct NoteWidgetEntry: TimelineEntry {
+    static let placeholder = NoteWidgetEntry(date: Date(),
+                                             title: DemoContent.singleNoteTitle,
+                                             content: DemoContent.singleNoteContent,
+                                             url: DemoContent.demoURL)
+
+    init(date: Date, note: Note) {
+        self.init(date: date,
+                  title: note.title,
+                  content: note.body,
+                  url: note.url)
+    }
+
+    init(date: Date, title: String, content: String, url: URL) {
+        self.date = date
+        self.title = title
+        self.content = content
+        self.url = url
+    }
+
+    let date: Date
+    let title: String
+    let content: String
+    let url: URL
+}
+
+struct NoteWidgetProvider: IntentTimelineProvider {
+    typealias Intent = NoteWidgetIntent
+    typealias Entry = NoteWidgetEntry
+
+    let coreDataManager: CoreDataManager!
+    let dataController: WidgetDataController!
+
+    init() {
+        do {
+            self.coreDataManager = try CoreDataManager(StorageSettings().sharedStorageURL, for: .widgets)
+            let isPreview = ProcessInfo.processInfo.environment[Constants.environmentXcodePreviewsKey] != Constants.isPreviews
+            self.dataController = try WidgetDataController(coreDataManager: coreDataManager, isPreview: isPreview)
+        } catch {
+            fatalError("Couldn't setup dataController")
+        }
+    }
+
+    func placeholder(in context: Context) -> NoteWidgetEntry {
+        return NoteWidgetEntry.placeholder
+    }
+
+    func getSnapshot(for configuration: NoteWidgetIntent, in context: Context, completion: @escaping (NoteWidgetEntry) -> Void) {
+        guard let dataController = dataController,
+              let note = try? dataController.firstNote() else {
```

Comment:
You may collapse the two first lines as follows (Unless I'm missing something!!)

```
guard let note = try? dataController?.firstNote() else {
```